### ISSUE TITLE
Is there any necessary to lock when update params?

### ISSUE THREAD

**Issue by Ranglage** [dgriff777/rl_a3c_pytorch#36]
It seems doesn't get lock when update network params in SharedAdam. 
However, isn't there process safety problem without a lock?

**Comment by dgriff777** [comment]
Yes, there is possibility for bad updates if race condition occurs but such bad updates are quickly overwritten by new incoming updates. Although this can look to be as a disadvantage to this method, the use of locks are substantially time consuming. Overall, any negative is far outweighed by the increase in speed of updates.
